x-pack/plugins/lens/public/editor_frame_service/editor_frame/state_helpers.ts

```diff
@@ -134,6 +148,50 @@ export async function persistedStateToExpression(
   };
 }
 
+export function getMissingIndexPattern(
+  currentDatasourceState:
+    | {
+        isLoading: boolean;
+        state: unknown;
+      }
+    | null
+    | string
+) {
+  if (currentDatasourceState == null || typeof currentDatasourceState === 'string') {
+    return [];
+  }
+  const datasourceState =
```

Comment:
We can't do this - this is blurring the line between frame and datasource in a non-typesafe way. What if we add a datasource later on which manages their internal state in a different way? Please introduce a new function `getIndexPatterns` on the `Datasource` interface which can be called here to obtain the list in a safe way.